gobblin-core/src/main/java/gobblin/instrumented/Instrumented.java

```diff
@@ -131,6 +142,92 @@ public static MetricContext getMetricContext(State state, Class<?> klazz, List<T
   }
 
   /**
+   * Generates a new {@link gobblin.metrics.MetricContext} with the parent and tags taken from the reference context.
+   * Allows replacing {@link gobblin.metrics.Tag} with new input tags.
+   * This method will not copy any {@link gobblin.metrics.Metric} contained in the reference {@link gobblin.metrics.MetricContext}.
+   *
+   * @param context Reference {@link gobblin.metrics.MetricContext}.
+   * @param newTags New {@link gobblin.metrics.Tag} to apply to context. Repeated keys will override old tags.
+   * @param name Name of the new {@link gobblin.metrics.MetricContext}.
+   *             If absent or empty, will modify old name by adding a random integer at the end.
+   * @return Generated {@link gobblin.metrics.MetricContext}.
+   */
+  public static MetricContext newContextFromReferenceContext(MetricContext context, List<Tag<?>> newTags,
+      Optional<String> name) {
+
+    String newName = name.orNull();
+
+    if (Strings.isNullOrEmpty(newName)) {
+      UUID uuid = UUID.randomUUID();
+      String randomIdPrefix = "uuid:";
+
+      String oldName = context.getName();
+      List<String> splitName = Strings.isNullOrEmpty(oldName) ?
+          Lists.<String>newArrayList() :
+          Splitter.on(".").splitToList(oldName);
```

Comment:
`splitToList` returns an immutable list. Can you change this? 
You can push to upstream without creating a new PR.